components/bl-modal-component/src/index.js

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from 'react';
+
+const { cn } = BackendlessUI.CSSUtils;
+const ESCAPE_KEY_CODE = 27;
+
+export default function ModalComponent({ component, eventHandlers, pods }) {
+  const { display, classList, style, disabled } = component;
+  const { onClose } = eventHandlers;
+
+  const [visibility, setVisibility] = useState(display);
+
+  component.openModal = () => setVisibility(true);
+  component.closeModal = () => {
+    setVisibility(false);
+    onClose({ visibility: false });
+  };
+
+  useEffect(() => {
+    setVisibility(display);
+  }, [display]);
+
+  const modalContentPod = pods['modalContent'];
+
+  useCloseOnEscape(onClose, setVisibility);
+
+  const handleClick = () => {
+    setVisibility(false);
+    onClose({ visibility: false });
+  };
+
+  if (!visibility) {
+    return null;
+  }
+
+  return (
+    <div
+      style={ style }
+      className={ cn("bl-customComponent-modal", classList, { "bl-customComponent-modal--disabled": disabled }) }>
```

Comment:
Do we really need "disabled" property? I can't imagine cases where we need to show modal with backdrop and disable all content in it.